demo/text-field-custom-demos.html

```diff
@@ -0,0 +1,78 @@
+<dom-module id="text-field-custom-demos">
+  <template>
+    <style include="vaadin-component-demo-shared-styles">
+      :host {
+        display: block;
+      }
+    </style>
+
+    <h3>Custom Input</h3>
+    <p>
+      <b>Note:</b> In order to apply lumo or material theme styles to the placeholder
+      of the slotted input you should explicitly import a file
+      `theme/lumo/vaadin-placeholder-styles.html` or `theme/material/vaadin-placeholder-styles.html`
+      and include styles into the same scope as `vaadin-text-field`:
+      <code>&lt;custom-style&gt;&lt;style include="lumo-placeholder-styles"&gt;&lt;/style&gt;&lt;/custom-style&gt;</code> OR 
+      <code>&lt;custom-style&gt;&lt;style include="material-placeholder-styles"&gt;&lt;/style&gt;&lt;/custom-style&gt;</code>
+    </p>
+    <vaadin-demo-snippet id="text-field-custom-demos-phone">
+      <template preserve-content>
+        <custom-style>
+          <style include="lumo-placeholder-styles"></style>
+        </custom-style>
+        <vaadin-text-field label="Name">
+          <input type="text" slot="input">
+        </vaadin-text-field>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Number Input</h3>
+    <p>
+      <b>Note:</b> All properties and attributes that `vaadin-text-field` has should be defined on the host.
+    </p>
+    <vaadin-demo-snippet id="text-field-custom-demos-phone">
+      <template preserve-content>
+        <vaadin-text-field label="Amount" pattern="^[1-9]{1,2}[0]?|100" prevent-invalid-input value="0">
+          <input type="number" slot="input" min="0" max="100" step="2">
```

Comment:
Wondering why the input allows any text on Safari. The "Prevent Invalid Input" example under "Validation" properly prevents typing letters on Safari also.